cmd/web-handlers.go

```diff
@@ -1452,8 +1452,30 @@ type PresignedGetRep struct {
 func (web *webAPIHandlers) PresignedGet(r *http.Request, args *PresignedGetArgs, reply *PresignedGetRep) error {
 	claims, owner, authErr := webRequestAuthenticate(r)
 	if authErr != nil {
+		if authErr == errNoAuthToken {
+			// Check if anonymous (non-owner) has access to download objects.
+			if !globalPolicySys.IsAllowed(policy.Args{
+				Action:          policy.GetObjectAction,
+				BucketName:      args.BucketName,
+				ConditionValues: getConditionValues(r, ""),
+				IsOwner:         false,
+				ObjectName:      args.ObjectName,
+			}) {
+				return toJSONError(errAuthentication)
+			}
+			reply.UIVersion = browser.UIVersion
+			reply.URL = args.HostName + s3utils.EncodePath("/"+path.Join(args.BucketName, args.ObjectName))
+			return nil
+		}
 		return toJSONError(authErr)
 	}
+
+	if args.BucketName == "" || args.ObjectName == "" {
+		return &json2.Error{
+			Message: "Bucket and Object are mandatory arguments.",
+		}
+	}
```

Comment:
This is a regression bug in the browser code, i.e browser should not call presignedGet when the user is not logged-in. Fix is:
```
diff --git a/browser/app/js/objects/actions.js b/browser/app/js/objects/actions.js
index 3268ecde..f96002e6 100644
--- a/browser/app/js/objects/actions.js
+++ b/browser/app/js/objects/actions.js
@@ -279,7 +279,7 @@ export const downloadObject = object => {
     } else {
       const url = `${
         window.location.origin
-      }${minioBrowserPrefix}/download/${currentBucket}/${encObjectName}?token=''`
+      }${minioBrowserPrefix}/download/${currentBucket}/${encObjectName}?token=`
       window.location = url
     }
   }
```

Reply:
oh @krishnasrinivas

Reply:
do you want to fix the code for zip also:
```
--- a/browser/app/js/objects/actions.js
+++ b/browser/app/js/objects/actions.js
@@ -308,7 +308,7 @@ export const downloadCheckedObjects = () => {
       objects: getCheckedList(state)
     }
     if (!web.LoggedIn()) {
-      const requestUrl = location.origin + "/minio/zip?token=''"
+      const requestUrl = location.origin + "/minio/zip?token="
       downloadZip(requestUrl, req, dispatch)
     } else {
       return web
```

Reply:
yes fixing all of it @krishnasrinivas @poornas